django_components/__init__.py

```diff
@@ -8,15 +8,14 @@
 from django.template.engine import Engine
 from django.utils.module_loading import autodiscover_modules
 
+from django_components.app_settings import app_settings
 from django_components.template_loader import Loader
 
 if django.VERSION < (3, 2):
     default_app_config = "django_components.apps.ComponentsConfig"
 
 
 def autodiscover():
-    from . import app_settings
-
```

Comment:
I hear you. I wasn't sure about this one either. We can put it back just in case.

At any rate this is how I reasoned about it:

1) If we stop treating the AppSettings instance as a faux-module  (see below), importing it instead, then by importing it at the top of \_\_init\_\_ there is a guarantee other modules can import settings as `from django_components import app_settings` or `from . import app_settings` (which IMO is pretty clean).
2) At some point app settings need to be obtained anyway, so we might as well get it over and done with quickly.
3) Counter to (2) – AppSettings breaks reference to settings if settings has no COMPONENTS attribute. Consequently, if for some reason django_components is imported (and \_\_init\_\_ is run) _before_ settings.configure() is run, then the settings are out of step. Perhaps this is what you were trying to avoid?

Reply:
I agree that your way is cleaner. Problem is, I didn't write this code myself, and I don't think the original author did it either. Ot was probably just copied it from something like allauth (that you link below). Maybe take the safer path, and focus on the slot/fill changes?

Reply:
Okay, probably safer to leave it inside the autodiscover call then. I'll make a to-do to figure out what this is doing.

As for avoiding sys.modules manipulation and not replacing app_settings.py by an instance of AppSettings, that's a harmless change. It really boils down to the same thing, except now my IDE no longer complains when I try to access a settings property. 

Does that sound alright?

Reply:
I trust you! :)